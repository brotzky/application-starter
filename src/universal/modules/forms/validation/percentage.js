export default function(percent, comparison) {
  return function(value) {
    if (!value) {
      return '';
    }
    const percentFloat = parseFloat(value.replace(/%/g, ''));
    if (comparison === 'max') {
      return percentFloat <= percent
        ? undefined
        : {
            defaultMessage: `cannot be greater than ${percent}%`,
            id: 'form.errors.percentage1',
            values: {},
          };
    } else {
      return percentFloat >= percent
        ? undefined
        : {
            defaultMessage: `cannot be less than ${percent}%`,
            id: 'form.errors.percentage1',
            values: {},
          };
    }
  };
}
