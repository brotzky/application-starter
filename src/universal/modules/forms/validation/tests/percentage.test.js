import percentage from '../percentage';

describe('[Form Validation] percentage', () => {
  describe('percentage()', () => {
    it('should return a function', () => {
      expect(percentage()).toMatchSnapshot();
    });

    it('should return an empty string if nothing is passed it', () => {
      expect(percentage()()).toMatchSnapshot();
    });

    it('should return no errors if a percentage less than 100 is passed in', () => {
      expect(percentage('100', 'max')('99%')).toMatchSnapshot();
    });

    it('should return no errors if a percentage greater than 100 is passed in', () => {
      expect(percentage('100', 'min')('101%')).toMatchSnapshot();
    });

    it('should return no errors if a percentage less than 100 is passed in', () => {
      expect(percentage('100', 'random')('101')).toMatchSnapshot();
    });

    it('should return an error when a percentage over 100 is passed in', () => {
      expect(percentage('100', 'max')('120')).toMatchSnapshot();
    });

    it('should return an error when a percentage less than 100 is passed in', () => {
      expect(percentage('100', 'min')('90')).toMatchSnapshot();
    });

    it('should return an error when a percentage less than 100 is passed in, with comparison string being a random string other than "max"', () => {
      expect(percentage('100', 'random')('99')).toMatchSnapshot();
    });

    it('should return no error when a percentage passed in equals to the percentage in comparison ("max" comparison)', () => {
      expect(percentage('91', 'max')('91')).toMatchSnapshot();
    });

    it('should return no error when a percentage passed in equals to the percentage in comparison ("min" comparison)', () => {
      expect(percentage('90', 'min')('90')).toMatchSnapshot();
    });

    it('should return no error when a percentage passed in equals to the percentage in comparison ("random" comparison)', () => {
      expect(percentage('80', 'random')('80')).toMatchSnapshot();
    });
  });
});
