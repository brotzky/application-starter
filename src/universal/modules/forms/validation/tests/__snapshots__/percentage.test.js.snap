// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Form Validation] percentage percentage() should return a function 1`] = `[Function]`;

exports[`[Form Validation] percentage percentage() should return an empty string if nothing is passed it 1`] = `""`;

exports[`[Form Validation] percentage percentage() should return an error when a percentage less than 100 is passed in 1`] = `
Object {
  "defaultMessage": "cannot be less than 100%",
  "id": "form.errors.percentage1",
  "values": Object {},
}
`;

exports[`[Form Validation] percentage percentage() should return an error when a percentage less than 100 is passed in, with comparison string being a random string other than "max" 1`] = `
Object {
  "defaultMessage": "cannot be less than 100%",
  "id": "form.errors.percentage1",
  "values": Object {},
}
`;

exports[`[Form Validation] percentage percentage() should return an error when a percentage over 100 is passed in 1`] = `
Object {
  "defaultMessage": "cannot be greater than 100%",
  "id": "form.errors.percentage1",
  "values": Object {},
}
`;

exports[`[Form Validation] percentage percentage() should return no error when a percentage passed in equals to the percentage in comparison ("max" comparison) 1`] = `undefined`;

exports[`[Form Validation] percentage percentage() should return no error when a percentage passed in equals to the percentage in comparison ("min" comparison) 1`] = `undefined`;

exports[`[Form Validation] percentage percentage() should return no error when a percentage passed in equals to the percentage in comparison ("random" comparison) 1`] = `undefined`;

exports[`[Form Validation] percentage percentage() should return no errors if a percentage greater than 100 is passed in 1`] = `undefined`;

exports[`[Form Validation] percentage percentage() should return no errors if a percentage less than 100 is passed in 1`] = `undefined`;

exports[`[Form Validation] percentage percentage() should return no errors if a percentage less than 100 is passed in 2`] = `undefined`;
